src/librustc/middle/stability.rs

```diff
@@ -614,8 +639,14 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         if let Some(&Stability{rustc_depr: Some(attr::RustcDeprecation { reason, since }), ..})
                 = stability {
             if let Some(id) = id {
+                let path = self.item_path_str(def_id);
+                let message = format!("use of deprecated item '{}'", path);
                 if deprecation_in_effect(&since.as_str()) {
```

Comment:
Sorry, I didn't notice before, but we need to make the same changes here. This warning is for deprecated items inside the standard library, etc. It should be affected by `DEPRECATED_IN_FUTURE` as well.